### ISSUE TITLE
Bucket Transitions non-array not detected

### ISSUE THREAD

**Issue by martysweet** [martysweet/cfn-lint#43]
```
  S3LoggingBucket:
    Type: AWS::S3::Bucket
    Properties:
      LifecycleConfiguration:
        Rules:
          - Status: Enabled
            Transitions:
                StorageClass: GLACIER
                TransitionInDays: 60
```

is Invalid as Transitions should be array types (http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig-rule.html).

```
  S3LoggingBucket:
    Type: AWS::S3::Bucket
    Properties:
      LifecycleConfiguration:
        Rules:
          - Status: Enabled
            Transitions:
              - StorageClass: GLACIER
                TransitionInDays: 60
```

would be correct...

**Comment by martysweet** [comment]
This appears to be a problem with the aws_resources document from AWS.
https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json

Below you can see the deprecated `Transition` and the newer `Transitions` property which should expect the list, however the specification for these is the same.

```json
        "Transition": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig-rule.html#cfn-s3-bucket-lifecycleconfig-rule-transition",
          "Required": false,
          "Type": "Transition",
          "UpdateType": "Mutable"
        },
        "Transitions": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig-rule.html#cfn-s3-bucket-lifecycleconfig-rule-transitions",
          "Required": false,
          "Type": "Transition",
          "UpdateType": "Mutable"
        }
```

The correct specification for this resource would be as below. The following specification would make the resource property be parsed as a list.

```json
"Transitions": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig-rule.html#cfn-s3-bucket-lifecycleconfig-rule-transitions",
          "Required": false,
          "Type": "Transition",
          "UpdateType": "Mutable"
          "ItemType": "Transition",
          "Type": "List"
        }
```
